pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,300 @@
+	// +optional
+	Verbs []string
+	// Scope is the target resource or the subresource the rule cares about. APIGroups, APIVersions, Resources, etc.
+	// Required.
+	Scope RequestScope
+}
+
+// UserInfo holds the information about the user needed to implement the
+// user.Info interface.
+type UserInfo struct {
+	// The name that uniquely identifies this user among all active users.
+	// +optional
+	Username *string
+	// A unique value that identifies this user across time. If this user is
+	// deleted and another user by the same name is added, they will have
+	// different UIDs.
+	// +optional
+	UID *string
+	// The names of groups this user is a part of.
+	// +optional
+	Groups []string
+	// Any additional information provided by the authenticator.
+	// +optional
+	Extra map[string]ExtraValue
+}
+
+// ExtraValue masks the value so protobuf can generate
+type ExtraValue []string
+
+// RequestScope
+type RequestScope struct {
+	// APIGroups is the API groups the resources belong to. '*' is all groups.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	APIGroups []string
+	// Resources is a list of resources this rule applies to.
+	//
+	// For example:
+	// 'pods' means pods.
+	// 'pods/log' means the log subresource of pods.
+	// '*' means all resources, but not subresources.
+	// 'pods/*' means all subresources of pods.
+	// '*/scale' means all scale subresources.
+	// '*/*' means all resources and their subresources.
+	//
+	// If wildcard is present, the validation rule will ensure resources do not
+	// overlap with each other.
+	//
+	// Depending on the enclosing object, subresources might not be allowed.
+	// Required.
+	Resources []string
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	// +optional
+	ResourceNames []string
+	// NonResourceURLs is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
+	// If an action is not a resource API request, then the URL is split on '/' and is checked against the NonResourceURLs to look for a match.
+	// Since non-resource URLs are not namespaced, this field is only applicable for ClusterRoles referenced from a ClusterRoleBinding.
+	// Rules can either apply to API resources (such as "pods" or "secrets") or non-resource URL paths (such as "/api"),  but not both.
+	// +optional
+	NonResourceURLs []string
+}
+
+// FlowSchemaStatus represents the current state of a flow-schema.
+type FlowSchemaStatus struct {
+	// Current state of flow-schema.
+	Conditions []FlowSchemaStatusCondition
+}
+
+// FlowSchemaStatusCondition describes conditions for a flow-schema.
+type FlowSchemaStatusCondition struct {
+	// Type is the type of the condition.
+	Type FlowSchemaConditionType
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status FlowSchemaConditionStatus
```

Comment:
Could we import and use `ConditionStatus` from `core/v1`?

Reply:
i'm fine w/ both. currently we're already referencing `core/v1` condition from `batch/v1` api, and i don't see strong difference between them.